interfaces/system_key.go

```diff
@@ -251,8 +252,45 @@ func SystemKeyMismatch() (bool, error) {
 	diskSystemKey.AppArmorParserFeatures = nil
 	mySystemKey.AppArmorParserFeatures = nil
 
+	ok, err := MatchSystemKeys(mySystemKey, &diskSystemKey)
+	return !ok, err
+}
+
+// RecordedSystemKey returns opaque system key read from the disk.
+func RecordedSystemKey() (interface{}, error) {
+	raw, err := ioutil.ReadFile(dirs.SnapSystemKeyFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, ErrSystemKeyMissing
+		}
+		return nil, err
+	}
+
+	var diskSystemKey systemKey
+	if err := json.Unmarshal(raw, &diskSystemKey); err != nil {
+		return nil, err
+	}
+
+	return &diskSystemKey, nil
+}
+
+// CurrentSystemKey calculates and returns opaque system key.
+func CurrentSystemKey() (interface{}, error) {
+	currentSystemKey, err := generateSystemKey()
+	return currentSystemKey, err
+}
+
+// MatchSystemKeys compares opaque system keys
+func MatchSystemKeys(systemKey1, systemKey2 interface{}) (bool, error) {
```

Comment:
Nitpick on the name, since this compares system keys for equality, how about just `SystemKeysEqual`